fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java

```diff
@@ -193,6 +199,11 @@ public void analyzeSelectClause() throws AnalysisException {
                 // Function must match pattern.
                 FunctionCallExpr functionCallExpr = (FunctionCallExpr) selectListItemExpr;
                 String functionName = functionCallExpr.getFnName().getFunction();
+                // current version not support count(distinct) function in creating materialized view
+                if (!isReplay && functionName.toLowerCase().equals("count") && functionCallExpr.isDistinct()) {
```

Comment:
There is really no problem with this repair. But I think a better way is:
1. Modify MVColumnOneChildPattern
2. ```  if (isReplay) {
                  ignore pattern
            } else {match pattern}
     ```